velox/expression/tests/ExprTest.cpp

```diff
@@ -760,6 +765,36 @@ TEST_F(ExprTest, csePartialEvaluation) {
   assertEqualVectors(expected, results[1]);
 }
 
+TEST_F(ExprTest, csePartialEvaluationWithEncodings) {
+  auto data = makeRowVector(
+      {wrapInDictionary(
+           makeIndicesInReverse(5),
+           wrapInDictionary(
+               makeIndicesInReverse(5),
+               makeFlatVector<int64_t>({0, 10, 20, 30, 40}))),
+       makeFlatVector<int64_t>({3, 33, 333, 3333, 33333})});
+
+  auto exprSet = compileMultiple(
+      {
+          "concat(concat(cast(c0 as varchar), ',', cast(c1 as varchar)), 'xxx')",
+          "concat(concat(cast(c0 as varchar), ',', cast(c1 as varchar)), 'yyy')",
+      },
+      asRowType(data->type()));
+
+  std::vector<VectorPtr> result(2);
+  {
+    SelectivityVector rows(2);
```

Comment:
Interesting. I think the fuzzer always creates the selectivity vector based on the input vector size. Is this valid actual valid input (smaller selectivity vector)?

If so, I guess we should add this logic to the fuzzer as well.

Reply:
@laithsakka haven't you recently added similar logic there (fuzzer selectivity vector input)?

Reply:
> I think the fuzzer always creates the selectivity vector based on the input vector size. Is this valid actual valid input (smaller selectivity vector)?

It is valid to request evaluation only on a subset of rows. These rows don't even have to be contiguous.

Reply:
Just to add context, this error was triggered when different inputs of different sizes passed through the same exprSet, which can happen if there is a filter in the scan node before the project node